code/go/0chain.net/sharder/sharder/sharder.go

```diff
@@ -224,9 +224,9 @@ func main() {
 	go sc.StartLFMBWorker(ctx)
 
 	setupBlockStorageProvider(mConf, workdir)
-	if sc.GetCurrentRound() == 0 {
-		sc.SetupGenesisBlock(viper.GetString("server_chain.genesis_block.id"),
-			magicBlock, initStates)
+	latestFinalizedBlock := sc.GetLatestFinalizedBlock()
+	if latestFinalizedBlock == nil || latestFinalizedBlock.Round == 0 {
+		sc.SetupGenesisBlock(viper.GetString("server_chain.genesis_block.id"), magicBlock, initStates)
```

Comment:
LatestFinalizedBlock() might work for sharder(need to confirm), but we also need to check the miners, whose LatestFinalizedBlock() seems always nil when on starting.

Reply:
Added LFB check for miners